File: src/System/System.Tests/System/Collections/ICollectionTests.cs
using Xunit;
using System.Collections.Generic;
namespace System.Collections
{
    /// <summary>
    /// ICollectionTests
    /// </summary>
    public class ICollectionTests
    {
        [Fact]
        public void Add_TwoElementList_CountEqualsThree()
        {
            var collection = (ICollection<int>)new List<int>(new int[] { 0, 1 });
            collection.Add(2);
            Assert.Equal(3, collection.Count);
        }

        [Fact]
        public void Clear_TwoElementList_CountEqualsZero()
        {
            var collection = (ICollection<int>)new List<int>(new int[] { 0, 1 });
            collection.Clear();
            Assert.Equal(0, collection.Count);
        }

        [Fact]
        public void Contains_TwoElementList_EqualsTrue()
        {
            var collection = (ICollection<int>)new List<int>(new int[] { 0, 1 });
            Assert.True(collection.Contains(1));
        }

        [Fact]
        public void CopyTo_TwoElementList_CountEqualsTrue()
        {
            var collection = (ICollection<int>)new List<int>(new int[] { 0, 1 });
            int[] array = new int[2];
            collection.CopyTo(array, 0);
            Assert.Equal(2, collection.Count);
        }

        [Fact]
        public void Remove_TwoElementList_CountEqualsOne()
        {
            var collection = (ICollection<int>)new List<int>(new int[] { 0, 1 });
            collection.Remove(1);
            Assert.Equal(1, collection.Count);
        }

        [Fact]
        public void getCount_TwoElementList_EqualsTwo()
        {
            var collection = (ICollection<int>)new List<int>(new int[] { 0, 1 });
            Assert.Equal(2, collection.Count);
        }

        [Fact]
        public void getReadOnly_TwoElementList_EqualsFalse()
        {
            var collection = (ICollection<int>)new List<int>(new int[] { 0, 1 });
            Assert.False(collection.IsReadOnly);
        }
    }
}

File: src/Thunk/Thunk.cpu.Tests/System/Collections/ICollectionTests.cpp
using namespace Xunit;
using namespace System;

/*
Declare_Collections_IEnumerator(System_Int32, int_)
Declare_Collections_IEnumerable(System_Int32, int_)
Declare_Collections_ICollection(System_Int32, int_)

array_Declare_GetEnumerable(System_Int32, int_)
Declare_System_Array(System_Int32, int_)
Declare_System_Action(System_Int32, int_)
Declare_System_Predicate(System_Int32, int_)
Declare_System_Comparison(System_Int32, int_)
Declare_Collections_IComparer(System_Int32, int_)
Declare_Collections_List(System_Int32, int_)

Declare_System_ArrayXSZArrayEnumerator(System_Int32, int_)
public bool MoveNext(System_ArrayXSZArrayEnumerator(System_Int32) *t)
{
printf("1a: %d, b: %d\n", t->_index, t->_endIndex);
if (t->_index < t->_endIndex)
{
t->_index++;
return (t->_index < t->_endIndex);
}
return false;
}
*/

namespace System { namespace Collections {
	/// <summary>
	/// ICollectionTests
	/// </summary>
	public ref class ICollectionTests
	{
	public:
		[Fact]
		void Add_TwoElementList_CountEqualsThree()
		{
			/*
			Collections_IEnumerable(System_Int32) *collection;
			Collections_IEnumerator(System_Int32) *enumerator;
			localArray(3, int_ ints[] = { 0, 1 });
			collection = array_getEnumerable(System_Int32, ints);
			enumerator = vcallT(Collections_IEnumerable,System_Int32,collection,GetEnumerator);
			enumerator->__vtable[0] = (System_ObjectVector)&MoveNext;
			while (vcallT(Collections_IEnumerator,System_Int32,enumerator,MoveNext) == true)
			printf("%d\n", getVcallT(Collections_IEnumerator,System_Int32,enumerator,Current));
			//b = array_getEnumerator(System_Int32, a);
			//collection = new(2,Collections_List(System_Int32), array_getEnumerable(System_Int32,ints));
			//vcallT(Collections_ICollection,System_Int32,collection,Add, 2);
			//NUnit_Assert_iAreEqual(3, getVcallT(Collections_ICollection,System_Int32,collection,Count));
			*/
		}

		[Fact]
		void Clear_TwoElementList_CountEqualsZero()
		{
		}

		[Fact]
		void Contains_TwoElementList_EqualsTrue()
		{
		}

		[Fact]
		void CopyTo_TwoElementList_CountEqualsTrue()
		{
		}

		[Fact]
		void Remove_TwoElementList_CountEqualsOne()
		{
		}

		[Fact]
		void getCount_TwoElementList_EqualsTwo()
		{
		}

		[Fact]
		void getReadOnly_TwoElementList_EqualsFalse()
		{
		}

	};
}}


